packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -641,6 +641,51 @@ describeWithDOM('mount', () => {
 </SuspenseComponent>`);
     });
 
+    it('return wrapped component when given loaded lazy component in initial mount', () => {
+      const LazyComponent = lazy(() => fakeDynamicImport(DynamicComponent));
+      /* eslint-disable no-underscore-dangle */
+      LazyComponent._result = DynamicComponent;
+      /* eslint-disable no-underscore-dangle */
+      LazyComponent._status = 1;
+      const SuspenseComponent = () => (
+        <Suspense fallback={<Fallback />}>
+          <LazyComponent />
+        </Suspense>
+      );
+
+      const wrapper = mount(<SuspenseComponent />);
+
+      expect(wrapper.is(SuspenseComponent)).to.equal(true);
+      expect(wrapper.find(LazyComponent)).to.have.lengthOf(0);
+      expect(wrapper.find(DynamicComponent)).to.have.lengthOf(1);
+      expect(wrapper.find(Fallback)).to.have.lengthOf(0);
+    });
+
+    it('return wrapped component string when given loaded lazy component in initial mount and call .debug()', () => {
+      const LazyComponent = lazy(() => fakeDynamicImport(DynamicComponent));
+      /* eslint-disable no-underscore-dangle */
+      LazyComponent._result = DynamicComponent;
+      /* eslint-disable no-underscore-dangle */
+      LazyComponent._status = 1;
+      const SuspenseComponent = () => (
+        <Suspense fallback={<Fallback />}>
+          <LazyComponent />
+        </Suspense>
+      );
+
+      const wrapper = mount(<SuspenseComponent />);
+
+      expect(wrapper.debug()).to.equal(`<SuspenseComponent>
```

Comment:
Note that the lazy component node is disappeared in mount when its loaded (`_result` get loaded and `_status` = 1). And that will make the `.debug()` different between shallow and mount when rendering `Suspense`. I'm not sure if we should also ignore the loaded lazy component node when shallow rendering `Suspense` to make behavior consistent.

Reply:
can you elaborate on this?

I would expect that a Suspense is in one of two states: render the children, or render the fallback, and that shallow and mount would be consistent for each of these states.

Reply:
For example, the following jsx
```js
const LazyComponent = React.lazy(() => import('WrappedComponent'))

<Suspense fallback={<Fallback />}>
  <LazyComponent />
</Suspense>
```
in mount, react dev tool shows either fallback when it's not loaded:

```js
<Suspense fallback={...}>
  <Fallback />
</Suspense>
```

or the wrapped component

```js
<Suspense fallback={...}>
  <WrappedComponent />
</Suspense>
```

In both cases the `LazyComponent` node is not shown.

But in shallow with `options.suspenseFallback = false` we will keep the `LazyComponent` node there no matter it's loaded or not because (I think) we might want to check if the lazy component node exists. And this makes it inconsistent with mount.

Reply:
We shouldn't need to mutate underscore-prefixed things to be able to test this.

Reply:
I agree that it's not good to use `_result` to test this. I'd try to [pass synchronous thenable](https://github.com/facebook/react/pull/14626) to replace `fakeDynamicImport` to see if that works.